include/pybind11/cast.h

```diff
@@ -960,9 +960,15 @@ template <typename type> class type_caster<std::reference_wrapper<type>> {
 private:
     using caster_t = make_caster<type>;
     caster_t subcaster;
-    using subcaster_cast_op_type = typename caster_t::template cast_op_type<type>;
-    static_assert(std::is_same<typename std::remove_const<type>::type &, subcaster_cast_op_type>::value,
-            "std::reference_wrapper<T> caster requires T to have a caster with an `T &` operator");
+    using reference_t = typename std::add_lvalue_reference<type>::type;
```

Comment:
Why not just `type&`? Apparently the only difference is `void`, but I don't think `std::reference_wrapper<void>` is supported, or is it?

Reply:
To be more specific, `type_caster<std::reference_wrapper<void>>` would already result in a compilation error a few lines further, to evaluate `std::is_same<typename std::remove_const<type>::type &, subcaster_cast_op_type>::value`: `typename std::remove_const<void>::type` is still `void`, and would also result in `void &`.

Reply:
Done.